snowfakery/plugins.py

```diff
@@ -74,9 +80,20 @@ def context_vars(self):
             self.plugin.__class__.__name__
         )
 
-    def evaluate(self, field_definition):
+    def evaluate_raw(self, field_definition):
+        """Evaluate the contents of a field definition"""
         return field_definition.render(self.interpreter.current_context)
 
+    def evaluate(self, field_definition):
+        """Evaluate the contents of a field definition and simplify to a primitive value."""
+        rc = self.evaluate_raw(field_definition)
+        if isinstance(rc, Scalar.__args__):
```

Comment:
Will this be a problem if rc is None, or is that not possible?

Reply:
The Scalar datatype does include None and None is basically a valid value in most contexts of Snowfakery. Outputstreams can output "null" or ,, or whatever is appropriate.